src/front/domain_socket.rs

```diff
@@ -84,6 +92,95 @@ impl DomainSocketListener {
 
         Ok(Self { listener, timeout })
     }
+
+    #[cfg(not(feature = "no-parsec-user-and-clients-group"))]
+    fn set_socket_dir_permissions(parent_dir: &Path) -> Result<()> {
+        if let Some(parent_dir_str) = parent_dir.to_str() {
+            fs::set_permissions(parent_dir, Permissions::from_mode(0o750))?;
+            // Although `parsec` has to be part of the `parsec_clients` group, it may not be the primary group. Therefore force group ownership to `parsec_clients`
+            if unsafe {
+                let parent_dir_cstr = CString::new(parent_dir_str)
+                    .expect("Failed to convert socket path parent to cstring");
+                {
+                    libc::chown(
+                        parent_dir_cstr.as_ptr(),
+                        users::get_current_uid(), // To get to this point, user has to be `parsec`
+                        users::get_group_by_name(PARSEC_GROUPNAME).unwrap().gid(), // `parsec_clients` exists by this point so should be safe
+                    )
+                }
+            } != 0
+            {
+                error!(
+                    "Changing ownership of {} to user {} and group {} failed.",
+                    parent_dir_str, PARSEC_USERNAME, PARSEC_GROUPNAME
+                );
+                return Err(Error::new(
+                    ErrorKind::Other,
+                    "Changing ownership of socket directory failed",
+                ));
+            }
+        } else {
+            error!(
+                "Error converting {} parent directory to string.",
+                SOCKET_PATH
+            );
+            return Err(Error::new(
+                ErrorKind::InvalidInput,
+                "Error retrieving parent directory for socket",
+            ));
+        }
+        Ok(())
+    }
+
+    #[cfg(not(feature = "no-parsec-user-and-clients-group"))]
+    fn check_user_details() -> Result<()> {
+        // Check Parsec is running as parsec user
+        if users::get_current_username() != Some(PARSEC_USERNAME.into()) {
+            error!(
+                "Incorrect user. Parsec should be run as user {}.",
+                PARSEC_USERNAME
+            );
+            return Err(Error::new(
+                ErrorKind::PermissionDenied,
+                "Parsec run as incorrect user",
+            ));
+        }
+        // Check Parsec client group exists and parsec user is a member of it
```

Comment:
Does `parsec` need to be part of `parsec-clients`? Since `parsec` owns the socket dir, can't they use it how they want?

Reply:
A user has to be a member of a group in order to change permissions of the group to that group. So if `parsec` is not a member of `parsec-clients`, it cannot change the group permission of the directory to `parsec-clients`.

Reply:
I see, that makes sense, thanks.